### PR TITLE
Created DoctrineObjectConstructor class

### PR DESCRIPTION
**Pull request by vlady777**
Created DoctrineObjectConstructor with additional logic to understand json:api format

### CODE REVIEW

File: Serializer/Construction/DoctrineObjectConstructor.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                default:
+                    throw new InvalidArgumentException("The provided fallback strategy for the object constructor is not valid");
+            }
+        }
+
+        $objectManager->initializeObject($object);
+
+        return $object;
+    }
+}
```
Comment (dbalabka) [inline Serializer/Construction/DoctrineObjectConstructor.php]: @vlady777 new line

File: Serializer/Construction/DoctrineObjectConstructor.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        $this->fallbackConstructor = $fallbackConstructor;
+        $this->fallbackStrategy = $fallbackStrategy;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function construct(VisitorInterface $visitor, ClassMetadata $metadata, $data, array $type, DeserializationContext $context)
+    {
+        $oldData = $data;
```
Comment (dbalabka) [inline Serializer/Construction/DoctrineObjectConstructor.php]: @vlady777 better name is `$jsonApiData`

File: Serializer/Construction/DoctrineObjectConstructor.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+ *
+ * @copyright 2018 OpticsPlanet, Inc
+ * @author    Vlad Yarus <vladislav.yarus@intexsys.lv>
+ */
+class DoctrineObjectConstructor implements ObjectConstructorInterface
+{
+    const ON_MISSING_NULL = 'null';
+    const ON_MISSING_EXCEPTION = 'exception';
+    const ON_MISSING_FALLBACK = 'fallback';
+    /**
```
Comment (dbalabka) [inline Serializer/Construction/DoctrineObjectConstructor.php]: @vlady777 new line

File: Serializer/Construction/DoctrineObjectConstructor.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        // Managed entity, check for proxy load
+        if (!is_array($data)) {
+            // Single identifier, load proxy
+            return $objectManager->getReference($metadata->name, $data);
+        }
+
+        // Fallback to default constructor if missing identifier(s)
+        $classMetadata = $objectManager->getClassMetadata($metadata->name);
+        if ($isJsoApi) {
+            $data[$classMetadata->getIdentifierFieldNames()[0]] = $oldData['data']['id'];
```
Comment (dbalabka) [inline Serializer/Construction/DoctrineObjectConstructor.php]: @vlady777 please add a comment that we are not supporting Composite Primary Keys. Probably we should throw an exception if there are more then 1 ID field names

**Review by dbalabka — CHANGES_REQUESTED** [review]
@vlady777 please take a look on PR comments

File: Serializer/Construction/DoctrineObjectConstructor.php (hunk 4 above)
Comment (vlady777) [reply Serializer/Construction/DoctrineObjectConstructor.php]: This comment already added in the class docblock

File: Serializer/Construction/DoctrineObjectConstructor.php (hunk 1 above)
Comment (vlady777) [reply Serializer/Construction/DoctrineObjectConstructor.php]: Added

File: Serializer/Construction/DoctrineObjectConstructor.php (hunk 2 above)
Comment (vlady777) [reply Serializer/Construction/DoctrineObjectConstructor.php]: Remaned

**Review by dbalabka — APPROVED** [review]

**Pull request merged**